leshan-server-cluster/src/main/java/org/eclipse/leshan/server/cluster/RedisSecurityStore.java

```diff
@@ -106,7 +106,14 @@ public SecurityInfo add(SecurityInfo info) throws NonUniqueSecurityInfoException
                 }
                 j.hset(PSKID_SEC.getBytes(), info.getIdentity().getBytes(), info.getEndpoint().getBytes());
             }
-            j.set((SEC_EP + info.getEndpoint()).getBytes(), data);
+
+            byte[] previousData = j.getSet((SEC_EP + info.getEndpoint()).getBytes(), data);
+            SecurityInfo previous = previousData == null ? null : deserialize(previousData);
+            String previousIdentity;
+            if (previous != null && (previousIdentity = previous.getIdentity()) != null
```

Comment:
Same thing about variable initialization here. 
If you really want to avoid to call several time `previous.getIdentity()` several time, you will probably need to create 2 if blocks. Personally I'm ok if you choose to not put `previous.getIdentity()` in a variable.